### PR TITLE
Include optional logger name in console output if it fails to load

### LINKED ISSUE #9289: [Bug]: The specified logger could not be created and will not be used. The given assembly name was invalid.
### Issue Description

I'm running dotnet locally, and it prints annoying warning without giving any details which logger failed

`The specified logger could not be created and will not be used. The given assembly name was invalid.`

dotnet (and msbuild) should print name of offending logger to help root cause the problem

### Steps to Reproduce

It's hard to tell which steps need to be taken to get into this state. On my machine it reproduces with simple 'dotnet new webapp'.

```
PS C:\src\Repo\src\myapp> dotnet new webapp
The template "ASP.NET Core Web App" was created successfully.
This template contains technologies from parties other than Microsoft, see https://aka.ms/aspnetcore/7.0-third-party-notices for details.

Processing post-creation actions...
Restoring C:\src\Repo\src\myapp\myapp.csproj:
The specified logger could not be created and will not be used. The given assembly name was invalid.
....
```

### Expected Behavior

Print name of logger that failed to load

### Actual Behavior

Logger name is not displayed

### Analysis

_No response_

### Versions & Configurations

_No response_

### PR DESCRIPTION
**Pull request by yanrez**
Include optional logger name in console output if it fails to load during initialization. 

Help identifying the logger that has issue

Fixes #9289

### Context
Currently if failure happens during initialization for optional logger, the message like this is printed:

`
The specified logger could not be created and will not be used. The given assembly name was invalid.
` 

It's very hard to debug since it doesn't give any hints which logger is having problems

### CODE REVIEW

File: src/MSBuild/XMake.cs [hunk 1]
```diff
@@ -4269,7 +4269,7 @@ private static LoggerDescription ParseLoggingParameter(string parameter, string
             }
             catch (Exception e) when (loggerDescription.IsOptional)
             {
-                Console.WriteLine(ResourceUtilities.FormatResourceStringStripCodeAndKeyword("OptionalLoggerCreationMessage", e.Message));
+                Console.WriteLine(loggerDescription.Name ": " + ResourceUtilities.FormatResourceStringStripCodeAndKeyword("OptionalLoggerCreationMessage", e.Message));
```
Comment (rainersigwald) [inline src/MSBuild/XMake.cs:4272]: Could you change the string itself, rather than prepending this?

Comment (yanrez) [reply src/MSBuild/XMake.cs]: Exception thrown is generic "The given assembly name was invalid." and resource string is generic localized string ("something went wrong"). So there is no venue to include details of loggerDescription .
I can add separate variable formatting the line before invoking Console.WriteLine(). I'm not familiar with codebase enough, not sure if there is preferred style.

Comment (rainersigwald) [reply src/MSBuild/XMake.cs]: I meant change the resource string--pushed a commit that does that to look at.

File: src/MSBuild/XMake.cs [hunk 2]
```diff
@@ -4269,7 +4269,7 @@ private static LoggerDescription ParseLoggingParameter(string parameter, string
             }
             catch (Exception e) when (loggerDescription.IsOptional)
             {
-                Console.WriteLine(ResourceUtilities.FormatResourceStringStripCodeAndKeyword("OptionalLoggerCreationMessage", e.Message));
```
Comment (danmoseley) [inline src/MSBuild/XMake.cs:4272]: What about all the messages from the catch blocks above - also for logger creation failed?

Comment (rainersigwald) [reply src/MSBuild/XMake.cs:4272]: Yes, we should feed it in there too.

**Comment by rainersigwald** [comment]
Looks like `Name` is not a good field to use here--it's empty in the tests we're running which seem pretty straightforward

```
❯ dotnet S:\msbuild\artifacts\bin\bootstrap\net8.0\MSBuild\MSBuild.dll -logger:",asdf.dll,IsOptional" .\MSBuild.Dev.slnf

MSBuild version 17.9.0-dev-23479-01+3aa07b74d for .NET
The specified logger "" could not be created and will not be used. Could not load file or assembly 'asdf.dll, Culture=neutral, PublicKeyToken=null'. The system cannot find the file specified.
```

**Review by AR-May — APPROVED** [review]

**Review by rainersigwald — APPROVED** [review]

File: src/MSBuild/XMake.cs (hunk 2 above)
Comment (JanKrivanek) [reply src/MSBuild/XMake.cs:4278]: @yanrez - do you plan to address this? 
Once it's adressed then this change is ready to go.

Comment (rainersigwald) [reply src/MSBuild/XMake.cs:4278]: I filed a bug to track this and don't think it's blocking this PR.

**Pull request merged**